program/src/processor.rs

```diff
@@ -0,0 +1,485 @@
+        *lookup_table_info.try_borrow_mut_data()?,
+        lookup_table_meta,
+    )?;
+
+    lookup_table_info.realloc(new_table_data_len, false)?;
+
+    {
+        let mut lookup_table_data = lookup_table_info.try_borrow_mut_data()?;
+        let uninitialized_addresses = AddressLookupTable::deserialize_addresses_from_index_mut(
+            &mut lookup_table_data,
+            old_table_data_len,
+        )?;
+        uninitialized_addresses.copy_from_slice(&new_addresses);
+    }
+
+    let rent = <Rent as Sysvar>::get()?;
+    let required_lamports = rent
+        .minimum_balance(new_table_data_len)
+        .max(1)
+        .saturating_sub(lookup_table_info.lamports());
+
+    if required_lamports > 0 {
+        let payer_info = next_account_info(accounts_iter)?;
+        let _system_program_info = next_account_info(accounts_iter)?;
+
+        if !payer_info.is_signer {
+            msg!("Payer account must be a signer");
+            return Err(ProgramError::MissingRequiredSignature);
+        }
+
+        invoke(
+            &system_instruction::transfer(payer_info.key, lookup_table_info.key, required_lamports),
+            &[payer_info.clone(), lookup_table_info.clone()],
+        )?;
+    }
+
+    Ok(())
+}
+
+fn process_deactivate_lookup_table(program_id: &Pubkey, accounts: &[AccountInfo]) -> ProgramResult {
+    let accounts_iter = &mut accounts.iter();
+
+    let lookup_table_info = next_account_info(accounts_iter)?;
+    let authority_info = next_account_info(accounts_iter)?;
+
+    if lookup_table_info.owner != program_id {
+        return Err(ProgramError::InvalidAccountOwner);
+    }
+
+    if !authority_info.is_signer {
+        msg!("Authority account must be a signer");
+        return Err(ProgramError::MissingRequiredSignature);
+    }
+
+    let mut lookup_table_meta = {
+        let lookup_table_data = lookup_table_info.try_borrow_data()?;
+        let lookup_table = AddressLookupTable::deserialize(&lookup_table_data)?;
+
+        if lookup_table.meta.authority.is_none() {
+            msg!("Lookup table is frozen");
+            // [Core BPF]: TODO: Should be `ProgramError::Immutable`
+            // See https://github.com/solana-labs/solana/pull/35113
+            return Err(ProgramError::Custom(0));
+        }
+        if lookup_table.meta.authority != Some(*authority_info.key) {
+            // [Core BPF]: TODO: Should be `ProgramError::IncorrectAuthority`
+            // See https://github.com/solana-labs/solana/pull/35113
+            return Err(ProgramError::Custom(0));
+        }
+        if lookup_table.meta.deactivation_slot != Slot::MAX {
+            msg!("Lookup table is already deactivated");
+            return Err(ProgramError::InvalidArgument);
+        }
+
+        lookup_table.meta
+    };
+
+    let clock = <Clock as Sysvar>::get()?;
+    lookup_table_meta.deactivation_slot = clock.slot;
+
+    AddressLookupTable::overwrite_meta_data(
+        *lookup_table_info.try_borrow_mut_data()?,
```

Comment:
Just for safety, let's use a new slice
```suggestion
        &mut lookup_table_info.try_borrow_mut_data()?[..],
```